lib/chef/provider/group/dscl.rb

```diff
@@ -88,8 +88,12 @@ def get_free_gid(search_limit = 1000)
 
         def gid_used?(gid)
           return false unless gid
-          groups_gids = safe_dscl("list", "/Groups", "gid")
-          !!( groups_gids =~ Regexp.new("#{Regexp.escape(gid.to_s)}\n") )
+          search_gids = safe_dscl("search", "/Groups", "PrimaryGroupID", gid.to_s)
+
+          # dscl -search should not return anything if the gid doesn't exist,
+          # but on the off-chance that it does, check whether the given gid is
+          # in the output.
+          !!(search_gids =~ /\b#{gid.to_s}\b/)
```

Comment:
`#{}` implies .to_s :) Since it's no longer wrapped in another method you should be able to drop .to_s